mifospay/src/main/java/org/mifos/mobilewallet/mifospay/merchants/presenter/MerchantsPresenter.java

```diff
@@ -93,5 +95,12 @@ public void onFailure(String message) {
             }
         });
 
+        Collections.sort(savingsWithAssociationsList, new Comparator<SavingsWithAssociations>() {
```

Comment:
Why are you doing the sorting here. This should be done in `onComplete` before passing the data to the UI.

Reply:
In the current state, you are setting incorrect `externalIds` to each item in the list. The list  on which `taskLooper` was doing its work will be different from the list on which you will be setting externId `savingsWithAssociationsList.get(index).setExternalId( responseValue.getClient().getExternalId());`
cause the list has been sorted in between. 

Let the taskLooper set externalId first to all the list items and then in `onComplete`, sort the final list

Reply:
@naman14 Updated according to the changes suggested.